homeassistant/components/sense/__init__.py

```diff
@@ -39,34 +43,90 @@
 )
 
 
-async def async_setup(hass, config):
-    """Set up the Sense sensor."""
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Sense component."""
+    hass.data.setdefault(DOMAIN, {})
+    conf = config.get(DOMAIN)
+    if not conf:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_EMAIL: conf.get(CONF_EMAIL),
```

Comment:
All these three fields are required or have a default and so are always available. Use `[…]` to indicate this to the reader of the code.

I don't know if this suggestion works. 

```suggestion
                CONF_EMAIL: conf[CONF_EMAIL],
                CONF_PASSWORD: [CONF_PASSWORD],
                CONF_TIMEOUT: conf[CONF_TIMEOUT],
```

Reply:
email and password are required. timeout is optional
will switch email and password to []